### ISSUE TITLE
How to process tasks on different workers without knowing the routing key (on a single REDIS instance)

### ISSUE THREAD

**Issue by pjmuller** [RichardKnop/machinery#637]
Hi,

first of all, amazing module! Works very well on a single worker in my current setup :).
Today I wanted to go to the next step and let separate `go` processes talk to each via machinery over one shared redis instance.

However I don't understand how the **default routing** works when you don't pass a `signature.RoutingKey` and have overlapping `DefaultQueue`'s.  Let me give a dummy example, imagine

process 1
```go
server, err := machinery.NewServer(redisCfg)
worker := server.NewWorker("worker", 100)
server.RegisterTask("task1", func1)
```

process 2
```go
server, err := machinery.NewServer(redisCfg)
worker := server.NewWorker("worker", 100)
server.RegisterTask("task2", func2)
result, err := server.SendTask(&tasks.Signature{Name: "task1", Args: []tasks.Arg{}})
```

task1 will be randomly processed by process 1 or 2 and thus only 50% of the time gets handled, whenever process 2 picks up task 1 it's not handled.

It does work every time if I 

1. pass a different `DefaultQueue` in the `redisCfg` for process 1 and 2
2. use the corresponding `signature.RoutingKey` 

but this reduces the flexibility as now process 2 needs to know the `RoutingKey` for task 1.

Is there a way to do it without setting  `DefaultQueue` and `signature.RoutingKey`?
So basically that by only using `RegisterTask` a server knows whether to leave a task for another server?

Thanks and happy holidays,
PJ

**Comment by pjmuller** [comment]
Possibly naive idea, what I register one server per task and give it the same `DefaultQueue` as the `taskQueue` it will accept?
Like that process 2 doesn't need to know about the Routing Key (as it's the same as the taskQueue)

However I'm not sure about the downsides of having +- 30 `servers` running on one process. Most likely they will all have a **separate** redis connection (to the same server), so not ideal...
